From fcad88b705fcc7eeb6aba4c753dd46e85ef1a509 Mon Sep 17 00:00:00 2001
From: Olliver Schinagl <oliver@schinagl.nl>
Date: Wed, 14 Aug 2024 10:30:34 +0200
Subject: [PATCH 06/14] sdk: Capture and forward all messages

When no endpoint found, default to endpoint 1. In the original source code
there is a return here. This prevent the messages from being forwarded. For
our use-case we want to capture all messages. Even if the coordinator is not
in the group.

@koenkk

Signed-off-by: Olliver Schinagl <oliver@schinagl.nl>
---
 source/ti/zstack/stack/af/af.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/source/ti/zstack/stack/af/af.c b/source/ti/zstack/stack/af/af.c
index 43b56f87a..72cd5fe24 100644
--- a/source/ti/zstack/stack/af/af.c
+++ b/source/ti/zstack/stack/af/af.c
@@ -433,10 +433,13 @@ void afIncomingData( aps_FrameFormat_t *aff, zAddrType_t *SrcAddress, uint16_t S
 #if !defined ( APS_NO_GROUPS )
     // Find the first endpoint for this group
     grpEp = aps_FindGroupForEndpoint( aff->GroupID, APS_GROUPS_FIND_FIRST );
-    if ( grpEp == APS_GROUPS_EP_NOT_FOUND )
-      return;   // No endpoint found
+    if ( grpEp == APS_GROUPS_EP_NOT_FOUND ) {
+      epDesc = afFindEndPointDesc( 1 );
+    }
+    else {
+      epDesc = afFindEndPointDesc( grpEp );
+    }
 
-    epDesc = afFindEndPointDesc( grpEp );
     if ( epDesc == NULL )
       return;   // Endpoint descriptor not found
 
-- 
2.46.0

